### DISCUSSION TITLE
Is there an equivalent to ‘Scenes’ from Telegraf?

### DISCUSSION THREAD

**Discussion by itsahedge** [mymmrac/telego#181] [Q&A] [answered]
I want to migrate a Telegram bot i'm building in JS using Telegraf npm pkg to using Golang, but I have some questions.

Does the idea of 'scenes' exist in TeleGo?

Basically the idea is to scope certain long running user interactions in scenes where you may be asking the user a bunch of steps in a command, storing data in context, using data from previous steps, etc.

You can read more here on scenes here:
https://stackoverflow.com/questions/61269477/how-do-scenes-and-stages-work-in-telegraf

**Comment by mymmrac** [comment, marked as answer]
Hello, unfortunately as of right now there is no such functionality in Telego, this is basically FSM (Finite State Machine), I discussed the reasoning why Telego doesn't have it and might not get it any time soon here (#125)
